crates/solver/src/driver.rs

```diff
@@ -377,6 +390,19 @@ impl Driver {
         };
 
         let mut settlement_transaction_attempted = false;
+        // In transition period last settlement is not necessarily the one with the
+        // highest score. So we need to get the scores of all settlements and
+        // sort them.
+        // CIP20 TODO - add to if statement below, once the transition period is over.
+        let mut scores = rated_settlements
+            .iter()
+            .map(|(solver, rated_settlement, _)| {
+                (solver.account().address(), rated_settlement.score.score())
+            })
+            .sorted_unstable_by(|(_, left_score), (_, right_score)| {
+                Ord::cmp(&right_score, &left_score) // descending
+            })
+            .peekable();
```

Comment:
nit: This iterator does not have to be peekable. You could do this to get both the solver and the score from it:
```rust
let (solver, score) = scores.next().expect("has a winner");
```

Reply:
Added.